cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocConf.java

```diff
@@ -421,11 +424,8 @@ static DataprocConf create(Map<String, String> properties) {
   static DataprocConf create(Map<String, String> properties, @Nullable SSHPublicKey publicKey) {
     String accountKey = getString(properties, "accountKey");
     if (accountKey == null || AUTO_DETECT.equals(accountKey)) {
-      try {
-        getComputeEngineCredentials();
-      } catch (IOException e) {
-        throw new IllegalArgumentException(e.getMessage(), e);
-      }
+      String endPoint = getString(properties, TOKEN_ENDPOINT_KEY);
+      new ComputeEngineCredentials(endPoint);
```

Comment:
What's the purpose of creating it here?

Reply:
According to comments, get credentials is being called here for validations:
_Create the conf from a property map while also performing validation._

So by calling it here, we make sure that token can be acquired.